internal/fsutil/create.go

```diff
@@ -18,8 +21,29 @@ type CreateOptions struct {
 	MakeParents bool
 }
 
+type Info struct {
+	Path string
+	Mode fs.FileMode
+	Hash string
+	Size uint
```

Comment:
The int type seems fine here, and avoids conversions. Note how the read/write functions take the signed type as well.